### PR TITLE
Added post a review and slack notification feature

### PR DESCRIPTION
**Pull request by AnthonyAmanse**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,18 @@ If you want to deploy the BookInfo app directly to Bluemix, click on 'Deploy to
...
+* MYSQL DB USER
+* MYSQL DB PASSWORD
+* MYSQL DB HOST  
+* MYSQL DB PORT
+
+> NOTE: Leave the MYSQL variables **blank** if you want to the toolchain to use MySQL in a container
+
+* SLACK WEBHOOK URL
+
+![images/env-variables.png](images/env-variables.png)
```
Comment (animeshsingh) [inline README.md]: I would remove the image here

File: .bluemix/toolchain.yml [hunk 2]
```diff
@@ -36,6 +36,11 @@ sample-build:
       API_KEY: "{{deploy.parameters.bluemix-api-key}}"
       BLUEMIX_ACCOUNT: "{{deploy.parameters.bluemix-cluster-account}}"
       CLUSTER_NAME: "{{deploy.parameters.bluemix-cluster-name}}"
+      MYSQL_DB_USER: "{{deploy.parameters.mysql-db-user}}"
+      MYSQL_DB_PASSWORD: "{{deploy.parameters.mysql-db-password}}"
+      MYSQL_DB_HOST: "{{deploy.parameters.mysql-db-host}}"
+      MYSQL_DB_PORT: "{{deploy.parameters.mysql-db-port}}"
+      SLACK_WEBHOOK_URL: "{{deploy.parameters.slack-webhook-url}}"
```
Comment (animeshsingh) [inline .bluemix/toolchain.yml]: I believe SLACK_WEBHOOK_URL is optional? No way to specify that?

File: README.md [hunk 3]
```diff
@@ -79,6 +98,8 @@ In this step, you can choose to build your Docker images from source in the [mic
...
+Another service called `post` was added that lets users who are logged in can post a review of their own. This service is using Node.js and it also needs access to the database. The service also sends data to the Slack webhook URL provided in `post-new.yaml` that sends a Slack message.
```
Comment (animeshsingh) [inline README.md]: Is this a new microservice or an extension to the bookinfo frontend service?

File: README.md [hunk 4]
```diff
@@ -87,6 +108,7 @@ The YAML files you need to modify are:
 * `details-new.yaml`
 * `reviews-new.yaml`
```
Comment (Tomcli) [inline README.md]: Maybe you can add a note that tells users to modify the env for all three versions of reviews.

**Comment by animeshsingh** [comment]
Ok I am in two minds whether we need a 'new' microservice. It could have been an extension to the existing frontend webpage microservice?

File: microservices/README.md [hunk 5]
```diff
@@ -43,9 +43,21 @@ $ docker build -t registry.ng.bluemix.net/<your-namespace>/reviews-v2
 $ docker build -t registry.ng.bluemix.net/<your-namespace>/reviews-v3
 ```
 
+## Product Page image
+
+1. Build the image using the Dockerfile in `productpage` folder.
+```bash
+$ dcoker build -t registry.ng.bluemix.net/<your-namespace>/productpage-v1
```
Comment (Tomcli) [inline microservices/README.md]: this should be `docker`

**Comment by paulczar** [comment]
I feel like we can kill this PR ?  I like the idea, and I think it would make a great blog post "extend istio to support slack notifications", but doesn't necessarily need to be here.

**Comment by animeshsingh** [comment]
That`s an idea we had - for now definitely in backburner

**Pull request closed without merging**